### PR TITLE
Prove +-comm for quo-int

### PR DESCRIPTION
**Pull request by ice1000**
As title.

### CODE REVIEW

File: Cubical/HITs/Ints/QuoInt/Properties.agda [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
++-pos-suc (posneg i) b = refl
++-pos-suc (neg (suc n)) b = simpleHcomp
+  (sym (sucPredℤ (pos b + neg n)))
+  (sym (predSucℤ (pos b + neg n)))
+  (cong predℤ (+-pos-suc (neg n) b))
+  -- the following hcomp does not termination-check:
+  -- hcomp (λ j → λ
+  --  { (i = i0) → sucPredℤ (pos b + neg n) (~ j)
+  --  ; (i = i1) → {!predℤ ((+-pos-suc (neg n) b) j)!}
+  --  }) (predSucℤ (pos b + neg n) (~ i))
```
Comment (ice1000) [inline Cubical/HITs/Ints/QuoInt/Properties.agda]: I suggest that Agda can do some enhancements here.

**Comment by ice1000** [comment]
Finally I've proved this after https://github.com/agda/cubical/issues/77#issuecomment-471741325

**Comment by ice1000** [comment]
A proof that didn't use Glue is still good to have -- we can say that the lemmas are useful for future proofs.

File: Cubical/Foundations/Prelude.agda [hunk 2]
```diff
@@ -61,6 +61,15 @@ cong₂ : ∀ {C : (a : A) → (b : B a) → Type ℓ} →
         PathP (λ i → C (p i) (q i)) (f x u) (f y v)
 cong₂ f p q i = f (p i) (q i)
 
+-- The simplest, one-dimensional hcomp
+-- sometimes needed to overcome a termination error
+
+simpleHcomp : ∀ {w x y z : A} → w ≡ z → w ≡ x → x ≡ y → z ≡ y
```
Comment (Rotsor) [inline Cubical/Foundations/Prelude.agda]: This is almost the same as `Cubical.Foundations.GroupoidLaws._∙∙_∙∙_` up to path directions, by the way.

Comment (ice1000) [reply Cubical/Foundations/Prelude.agda]: Writing down some notes:

https://github.com/agda/cubical/blob/7e6c71fa51b1633cd734e60eb8a6e7ba2707d4ea/Cubical/Foundations/GroupoidLaws.agda#L187-L188

Comment (mortberg) [reply Cubical/Foundations/Prelude.agda]: Yeah, I'm not sure we need this function as well then

File: Cubical/HITs/Ints/QuoInt/Properties.agda (hunk 1 above)
Comment (mortberg) [reply Cubical/HITs/Ints/QuoInt/Properties.agda]: This seems like a bug to me... @Saizan ?

File: Cubical/Foundations/Prelude.agda (hunk 2 above)
Comment (ice1000) [reply Cubical/Foundations/Prelude.agda]: Fixed

File: Cubical/HITs/Ints/QuoInt/Properties.agda (hunk 1 above)
Comment (Saizan) [reply Cubical/HITs/Ints/QuoInt/Properties.agda]: This is most likely an instance of https://github.com/agda/agda/issues/2100 , as pattern matching lambdas are essentially desugared to definitions in a where block.

Comment (ice1000) [reply Cubical/HITs/Ints/QuoInt/Properties.agda]: Well, it's a pity..

**Comment by ice1000** [comment]
Travis fails with:

```
/Data.o )

[335 of 373] Compiling Agda.TypeChecking.Rules.LHS ( src/full/Agda/TypeChecking/Rules/LHS.hs, dist/build/Agda/TypeChecking/Rules/LHS.o )

[336 of 373] Compiling Agda.TypeChecking.Coverage ( src/full/Agda/TypeChecking/Coverage.hs, dist/build/Agda/TypeChecking/Coverage.o )

No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received

The build has been terminated
```

which is not my fault

**Pull request merged**